azblob/zt_url_block_blob_test.go

```diff
@@ -899,3 +899,88 @@ func (s *aztestsSuite) TestBlobPutBlockListModifyBlob(c *chk.C) {
 	c.Assert(resp.UncommittedBlocks, chk.HasLen, 0)
 }
 
+func (s *aztestsSuite) TestSetTierOnBlobUpload(c *chk.C) {
+	bsu := getBSU()
+	containerURL, _ := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+	for _, tier := range []AccessTierType{AccessTierArchive, AccessTierCool, AccessTierHot} {
+		blobURL, _ := getBlockBlobURL(c, containerURL)
+
+		_, err := blobURL.Upload(ctx, strings.NewReader("Test Data"), basicHeaders, nil, BlobAccessConditions{}, tier)
+		c.Assert(err, chk.IsNil)
+
+		resp, err := blobURL.GetProperties(ctx, BlobAccessConditions{})
+		c.Assert(err, chk.IsNil)
+		c.Assert(resp.AccessTier(), chk.Equals, string(tier))
+	}
+}
+
+func (s *aztestsSuite) TestBlobSetTierOnCommit(c *chk.C) {
+	bsu := getBSU()
+	containerURL, _ := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+
+	for _, tier := range []AccessTierType{AccessTierCool, AccessTierHot} {
+		blobURL, _ := getBlockBlobURL(c, containerURL)
+
+		_, err := blobURL.StageBlock(ctx, blockID, strings.NewReader(blockBlobDefaultData), LeaseAccessConditions{}, nil)
+		c.Assert(err, chk.IsNil)
+
+		_, err = blobURL.CommitBlockList(ctx, []string{blockID}, BlobHTTPHeaders{}, nil, BlobAccessConditions{}, tier)
+
+		resp, err := blobURL.GetBlockList(ctx, BlockListCommitted, LeaseAccessConditions{})
+		c.Assert(err, chk.IsNil)
+		c.Assert(resp.CommittedBlocks, chk.HasLen, 1)
+		c.Assert(resp.UncommittedBlocks, chk.HasLen, 0)
+	}
+}
+
+func (s *aztestsSuite) TestSetTierOnCopyBlockBlobFromURL(c *chk.C) {
```

Comment:
maybe I missed it, but we need a test for the async copy method as well

Reply:
Sure. Will do that 👍